File: web/src/plugin/statistics/api/statistics.js
import service from '@/utils/request'

/**
 * 获取系统概览
 * @returns {Promise}
 */
export const getSystemOverview = () => {
  return service({
    url: '/statistics/systemOverview',
    method: 'get'
  })
}

/**
 * 获取CPU统计数据
 * @param {Object} params 查询参数
 * @param {string} params.timeRange 时间范围(如: now-1h, now-24h)
 * @param {string} params.interval 聚合间隔(如: 1m, 5m, 1h)
 * @param {string} params.hostName 主机名(可选)
 * @returns {Promise}
 */
export const getCPUStatistics = (params) => {
  return service({
    url: '/statistics/cpu',
    method: 'get',
    params: params
  })
}

/**
 * 获取内存统计数据
 * @param {Object} params 查询参数
 * @param {string} params.timeRange 时间范围
 * @param {string} params.hostName 主机名
 * @returns {Promise}
 */
export const getMemoryStatistics = (params) => {
  return service({
    url: '/statistics/memory',
    method: 'get',
    params: params
  })
}

/**
 * 获取文件完整性统计
 * @param {Object} params 查询参数
 * @param {string} params.timeRange 时间范围
 * @param {string} params.filePath 文件路径过滤
 * @param {string} params.eventType 事件类型(created/updated/deleted)
 * @param {string} params.hostName 主机名
 * @returns {Promise}
 */
export const getFileIntegrityStats = (params) => {
  return service({
    url: '/statistics/fileIntegrity',
    method: 'get',
    params: params
  })
}

/**
 * 获取进程统计数据
 * @param {Object} params 查询参数
 * @param {string} params.timeRange 时间范围
 * @param {string} params.processName 进程名过滤
 * @param {string} params.hostName 主机名
 * @param {number} params.topN Top N进程
 * @returns {Promise}
 */
export const getProcessStatistics = (params) => {
  return service({
    url: '/statistics/process',
    method: 'get',
    params: params
  })
}

/**
 * 获取审计事件统计
 * @param {Object} params 查询参数
 * @param {string} params.timeRange 时间范围
 * @param {string} params.category 事件类别
 * @param {string} params.action 动作
 * @param {string} params.userName 用户名
 * @param {string} params.hostName 主机名
 * @returns {Promise}
 */
export const getAuditEventStats = (params) => {
  return service({
    url: '/statistics/audit',
    method: 'get',
    params: params
  })
}

/**
 * 执行自定义ES查询
 * @param {Object} data 查询参数
 * @param {string} data.index 索引名称
 * @param {string} data.query 查询DSL(JSON格式)
 * @param {number} data.page 页码
 * @param {number} data.pageSize 每页数量
 * @returns {Promise}
 */
export const customQuery = (data) => {
  return service({
    url: '/statistics/customQuery',
    method: 'post',
    data: data
  })
}

File: server/plugin/statistics/model/request/statistics.go
package model

// ESSearchRequest ES搜索请求
type ESSearchRequest struct {
	Index     string `json:"index" form:"index"`         // 索引名称
	Query     string `json:"query" form:"query"`         // 查询DSL(JSON格式)
	Page      int    `json:"page" form:"page"`           // 页码
	PageSize  int    `json:"pageSize" form:"pageSize"`   // 每页数量
	TimeRange string `json:"timeRange" form:"timeRange"` // 时间范围
}

// CPUStatisticsRequest CPU统计请求
type CPUStatisticsRequest struct {
	TimeRange string `json:"timeRange" form:"timeRange"` // 时间范围(如: now-1h, now-24h)
	Interval  string `json:"interval" form:"interval"`   // 聚合间隔(如: 1m, 5m, 1h)
	HostName  string `json:"hostName" form:"hostName"`   // 主机名(可选)
}

// MemoryStatisticsRequest 内存统计请求
type MemoryStatisticsRequest struct {
	TimeRange string `json:"timeRange" form:"timeRange"`
	HostName  string `json:"hostName" form:"hostName"`
}

// FileIntegrityRequest 文件完整性统计请求
type FileIntegrityRequest struct {
	TimeRange string `json:"timeRange" form:"timeRange"`
	FilePath  string `json:"filePath" form:"filePath"`   // 文件路径过滤
	EventType string `json:"eventType" form:"eventType"` // 事件类型(created/updated/deleted)
	HostName  string `json:"hostName" form:"hostName"`
}

// ProcessStatisticsRequest 进程统计请求
type ProcessStatisticsRequest struct {
	TimeRange   string `json:"timeRange" form:"timeRange"`
	ProcessName string `json:"processName" form:"processName"` // 进程名过滤
	HostName    string `json:"hostName" form:"hostName"`
	TopN        int    `json:"topN" form:"topN"` // Top N进程
}

// AuditEventRequest 审计事件统计请求
type AuditEventRequest struct {
	TimeRange string `json:"timeRange" form:"timeRange"`
	Category  string `json:"category" form:"category"` // 事件类别
	Action    string `json:"action" form:"action"`     // 动作
	UserName  string `json:"userName" form:"userName"` // 用户名
	HostName  string `json:"hostName" form:"hostName"`
}
